File: DCAnalyticsWebShell/ClientApp/src/models/Question.ts
import { EnumList } from "./EnumList";
import { Answer } from "./Answer";
import { DataTypes, QuestionTypes } from "src/enums/enums";
import { SkipCondition } from "./SkipCondition";
import { Dependency } from "./Dependency";

export class Question {
  OID: number
  Key: string;
  Name: string;
  QuestionText: string;
  Deleted: boolean;
  DataType: DataTypes;
  QuestionType: QuestionTypes;
  CollectionObjectType: number;
  EnumList: EnumList;
  Answers: Answer[]
  Required: boolean = false
  Conditions: SkipCondition[]
  Dependencies: Dependency[]
  Occurance : number
}

File: DCAnalyticsMobile/DCAnalyticsMobile/Models/Question.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace DCAnalyticsMobile.Models
{
    public class Question : DataCollectionObject
    {
        public DCAnalytics.QuestionTypes QuestionType { get; set; }
        public string QuestionText { get; set; }
        public bool Required { get; set; }
        public bool Pending { get; set; }
        public DCAnalytics.DataTypes DataType { get; set; }
        public EnumList EnumList { get; set; }
        public List<Answer> Answers { get; set; }
        public List<SkipCondition> Conditions { get; set; }
        public List<Dependency> Dependencies  { get; set; }
        public int Occurance { get; set; }
    }
}
